presto-product-tests/src/main/java/io/prestosql/tests/cli/PrestoCliFileAuthTests.java

```diff
@@ -0,0 +1,190 @@
+import static io.prestosql.tests.TestGroups.FILEAUTH;
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertTrue;
+
+public class PrestoCliFileAuthTests
+        extends PrestoCliLauncher
+        implements RequirementsProvider
+{
+    @Inject(optional = true)
+    @Named("databases.presto.cli_truststore_path")
+    private String truststorePath;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_truststore_password")
+    private String truststorePassword;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_server_address")
+    private String serverAddress;
+
+    protected PrestoCliFileAuthTests()
+            throws IOException
+    {
+    }
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return compose(immutableTable(NATION));
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testCorrectPBKDF2Password()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userPBKDF2", "password", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testIncorrectPBKDF2Password()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userPBKDF2", "passwordPBKDF2", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testCorrectBCryptPassword()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testIncorrectBCryptPassword()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user1234", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testAuthTokenCacheUpdate()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+        //update the content in file and check for success
+        addContentToFile("/docker/volumes/conf/presto/etc/file_auth_copy", "userBCrypt1:$2y$10$BqTb8hScP5DfcpmHo5PeyugxHz5Ky/qf3wrpD7SNm8sWuA3VlGqsa");
+        Thread.sleep(10); // pausing till refresh-period for reloading the file
```

Comment:
Increase this value to `200`, since it will take some time to refresh the file, plus we need to account for thread scheduling.

Reply:
Done